### PR TITLE
Language info should be in the Session, not in the Connection

### PR DESCRIPTION
**Pull request by sco0ter**
I feel the language (the stream's "xml:lang" attribute) should be in the Session interface rather than the Connection (similar as the stream id), because it's unrelated to the session's transport.

The HttpSession already did that.

Otherwise the used language is not easy to retrieve, e.g. when using SessionManager to get the session and OF-208 is hard to solve.

Alternatively we can expose the Connection in the Session interface, but this approach feels wrong to me.

Also use `java.util.Locale` rather than String, because it better reflects what can be in `xml:lang` (it can also have country codes, e.g. "de-CH")

### CODE REVIEW

**Comment by dwd** [comment]
Looks good to me.

File: src/java/org/jivesoftware/openfire/session/Session.java [hunk 1]
```diff
@@ -212,4 +213,12 @@
      * @return cipher suite name.
      */
     public String getCipherSuiteName();
+
+    /**
+     * Returns the language code that should be used for this connection
```
Comment (tevans) [inline src/java/org/jivesoftware/openfire/session/Session.java]: Might prefer "session" here rather than "connection" (also below), but not a blocker.

**Comment by tevans** [comment]
+1 - thanks for including the plugins :)

**Comment by sco0ter** [comment]
Thanks, I've fixed the JavaDoc and also increased the WebSocket plugin minVersion because it requires the new `createClientSession(Connection, Locale);` API.

**Pull request merged**